lib/flip_card.dart

```diff
@@ -1,255 +1,303 @@
+    super.didUpdateWidget(oldWidget);
+
+    if (widget.duration != oldWidget.duration) {
+      controller.duration = widget.duration;
+    }
 
-  FlipCardState(this.isFront);
+    if (widget.controller != oldWidget.controller) {
+      if (oldWidget.controller?.state == this) {
+        oldWidget.controller?.state = null;
+      }
+
+      widget.controller?.state = this;
+    }
+  }
 
   @override
   void initState() {
     super.initState();
     controller = AnimationController(
-      value: isFront ? 0.0 : 1.0,
-      duration: Duration(milliseconds: widget.speed),
+      value: widget.initialSide == CardSide.front ? 0.0 : 1.0,
+      duration: widget.duration,
       vsync: this,
     );
-    _frontRotation = TweenSequence(
-      [
-        TweenSequenceItem<double>(
-          tween: Tween(begin: 0.0, end: pi / 2)
-              .chain(CurveTween(curve: Curves.easeIn)),
-          weight: 50.0,
-        ),
-        TweenSequenceItem<double>(
-          tween: ConstantTween<double>(pi / 2),
-          weight: 50.0,
-        ),
-      ],
-    ).animate(controller!);
-    _backRotation = TweenSequence(
-      [
-        TweenSequenceItem<double>(
-          tween: ConstantTween<double>(pi / 2),
-          weight: 50.0,
-        ),
-        TweenSequenceItem<double>(
-          tween: Tween(begin: -pi / 2, end: 0.0)
-              .chain(CurveTween(curve: Curves.easeOut)),
-          weight: 50.0,
-        ),
-      ],
-    ).animate(controller!);
 
     widget.controller?.state = this;
 
     if (widget.autoFlipDuration != null) {
-      Future.delayed(widget.autoFlipDuration!, toggleCard);
+      Future.delayed(widget.autoFlipDuration!, flip);
     }
   }
 
   @override
-  void didUpdateWidget(FlipCard oldWidget) {
-    widget.controller?.state ??= this;
+  void dispose() {
+    controller.dispose();
+    widget.controller?.state = null;
+    super.dispose();
   }
 
-  /// Flip the card
-  /// If awaited, returns after animation completes.
-  Future<void> toggleCard() async {
-    if(!mounted) return;
-    
+  /// {@template flip_card.FlipCardState.flip}
+  /// Flips the card or reverses the direction of the current animation
+  ///
+  /// You can optionally pass a [targetSide] to show next
+  ///
+  /// This function returns a future that will complete when animation is done
+  /// {@endtemplate}
+  Future<void> flip([CardSide? targetSide]) async {
+    if (!mounted) return;
     widget.onFlip?.call();
 
-    final isFrontBefore = isFront;
-    controller!.duration = Duration(milliseconds: widget.speed);
+    targetSide ??= getOppositeSide();
 
-    final animation = isFront ? controller!.forward() : controller!.reverse();
-    animation.whenComplete(() {
-      if (widget.onFlipDone != null) widget.onFlipDone!(isFront);
-      if (!mounted) return;
-      setState(() => isFront = !isFrontBefore);
-    });
+    switch (targetSide) {
+      case CardSide.front:
+        await controller.forward().complete;
```

Comment:
The flip() animation doesn't work at all.
However, it does work once I switch `forward()` with `reverse()` (line 174).
Maybe it's better to fix this in a different way though.

Reply:
I've fixed it